sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,151 @@
+
+        /// <summary>
+        /// The client for interaction with the Files Shares file system.
+        /// </summary>
+        private ShareClient FilesShareClient { get; set; }
+
+        /// <summary>
+        /// The path of the file that is created locally from the downloaded data.
+        /// </summary>
+        private string DownloadFilePath { get; set; }
+
+        /// <summary>
+        /// File uploader.
+        /// </summary>
+        private readonly UploadFile _fileUploader;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DownloadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public DownloadFile(SizeOptions options) : base(options)
+        {
+            _fileUploader = new UploadFile(options);
+            FilesShareClient = _fileUploader.FilesShareClient;
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to initialize and set up the environment for the test scenario.
+        /// When multiple instances are run in parallel, the setup will take place once, prior to the
+        /// execution of the first test instance.
+        /// </summary>
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            await _fileUploader.GlobalSetupAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to clean up the environment for the test scenario.
+        /// When multiple instances are run in parallel, the cleanup will take place once,
+        /// after the execution of all test instances.
+        /// </summary>
+        public override async Task GlobalCleanupAsync()
+        {
+            await base.GlobalCleanupAsync();
+
+            await _fileUploader.GlobalCleanupAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to initialize and set up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, setup will be run once for each prior to its execution.
+        /// </summary>
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            // Upload a file to the share storage. The test will use 'ShareClient' to download this file.
+            (FilesShareClient, DirectoryClientName, FileClientName) = await _fileUploader.UploadARandomFileAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to clean up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, cleanup will be run once for each after execution has completed.
+        /// </summary>
+        public override async Task CleanupAsync()
+        {
+            await base.CleanupAsync();
+
+            try
+            {
+                await _fileUploader.CleanupAsync();
```

Comment:
If you don't end up refactoring, you may want to consider whether you'd want a failure in the uploader cleanup to cause you to not attempt to delete the downloaded file.   May be worth putting them in their own `try/catch`